client/app/pages/queries/view.js

```diff
@@ -204,6 +204,25 @@ function QueryViewCtrl($scope, Events, $route, $routeParams, $location, $window,
     $scope.saveQuery(undefined, { is_draft: $scope.query.is_draft });
   };
 
+  $scope.saveAsDataset = () => {
```

Comment:
This can use the snazzy async await syntax, which you can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)